pkg/dataobj/metastore/object.go

```diff
@@ -109,7 +109,7 @@ func iterTableOfContentsPaths(start, end time.Time) iter.Seq2[string, multitenan
 func NewObjectMetastore(bucket objstore.Bucket, logger log.Logger, reg prometheus.Registerer) *ObjectMetastore {
 	store := &ObjectMetastore{
 		bucket:      bucket,
-		parallelism: 64,
+		parallelism: 16,
```

Comment:
Did you mean to commit this? (I don't have an opinion about whether 16 vs 64 is better, but you didn't mention it in the PR description so I'm just calling this out in case it was for local testing)